lib/rubocop/cop/lint/block_alignment.rb

```diff
@@ -53,9 +85,30 @@ def check_block_alignment(start_node, block_node)
           return unless begins_its_line?(end_loc)
 
           start_loc = start_node.source_range
-          return unless start_loc.column != end_loc.column
+          return unless start_loc.column != end_loc.column ||
+                        style == :start_of_block
+
+          do_source_line_column =
+            compute_do_source_line_column(block_node, end_loc)
+          return unless do_source_line_column
+
+          error_source_line_column = if style == :start_of_block
+                                       do_source_line_column
+                                     else
+                                       loc_to_source_line_column(start_loc)
+                                     end
+
+          fmt = format(
+            MSG,
+            format_source_line_column(loc_to_source_line_column(end_loc)),
+            format_source_line_column(error_source_line_column),
+            alt_start_msg(start_loc, do_source_line_column)
```

Comment:
Looks like you have gone out of your way to make the offense message very specific and informative. Nice.